### PR TITLE
Feature: add deprecated attribute

### PR DESCRIPTION
**Pull request by SSE4**
https://github.com/conan-io/conan/issues/7251
https://github.com/conan-io/conan/pull/7399

### CODE REVIEW

File: reference/conanfile/attributes.rst [hunk 1]
```diff
@@ -1454,3 +1454,44 @@ For example, a *conandata.yml* with information about sources that looks like th
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
+
+
+deprecated
+----------
+
+This attribute declares that recipe is deprecated, causing user-friendly warning message to be emitted whenever recipe is used.
```
Comment (memsharded) [inline reference/conanfile/attributes.rst:1462]: Make it experimental (warn notice)

File: reference/conanfile/attributes.rst [hunk 2]
```diff
@@ -1454,3 +1454,44 @@ For example, a *conandata.yml* with information about sources that looks like th
...
+        version = "1.0"
+        deprecated = "taskflow"
+
+will emit warning like:
+
+.. code-block:: bash
+
+    cpp-taskflow/1.0: WARN: Recipe 'cpp-taskflow/1.0' is deprecated in favor of 'taskflow'. Please, consider changing your requirements.
+
+If value of the attribute evaluates to ``False``, not warning is printed.
```
Comment (danimtb) [inline reference/conanfile/attributes.rst:1497]:
```suggestion
If the value of the attribute evaluates to ``False``, no warning is printed.
```

File: reference/conanfile/attributes.rst [hunk 3]
```diff
@@ -1454,3 +1454,44 @@ For example, a *conandata.yml* with information about sources that looks like th
...
+        version = "1.0"
+        deprecated = True
+
+may emit warning like:
+
+.. code-block:: bash
+
+    cpp-taskflow/1.0: WARN: Recipe 'cpp-taskflow/1.0' is deprecated. Please, consider changing your requirements.
+
+Optionally, the attribute may specify the name of the suggested replacement, e.g.:
```
Comment (danimtb) [inline reference/conanfile/attributes.rst:1480]:
```suggestion
Optionally, the attribute may specify the name of the suggested replacement:
```

File: reference/conanfile/attributes.rst [hunk 4]
```diff
@@ -1454,3 +1454,44 @@ For example, a *conandata.yml* with information about sources that looks like th
...
+.. code-block:: python
+
+    from conans import ConanFile
+
+    class Pkg(ConanFile):
+        name = "cpp-taskflow"
+        version = "1.0"
+        deprecated = "taskflow"
+
+will emit warning like:
```
Comment (danimtb) [inline reference/conanfile/attributes.rst:1491]:
```suggestion
This will emit a warning like:
```

File: reference/conanfile/attributes.rst [hunk 5]
```diff
@@ -1454,3 +1454,44 @@ For example, a *conandata.yml* with information about sources that looks like th
...
+.. code-block:: python
+
+    from conans import ConanFile
+
+    class Pkg(ConanFile):
+        name = "cpp-taskflow"
+        version = "1.0"
+        deprecated = True
+
+may emit warning like:
```
Comment (danimtb) [inline reference/conanfile/attributes.rst:1474]:
```suggestion
may emit a warning like:
```

File: reference/conanfile/attributes.rst (hunk 1 above)
Comment (danimtb) [inline reference/conanfile/attributes.rst:1462]:
```suggestion
This attribute declares that the recipe is deprecated, causing a user-friendly warning message to be emitted whenever it is used.
```

Comment (SSE4) [reply reference/conanfile/attributes.rst]: added

**Pull request merged**